src/transformState.js

```diff
@@ -5,7 +5,27 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(state, { ...action.extraData });
+        break;
+      case 'removeProperties':
+        const bro = action.keysToRemove;
+
+        for (const n of bro) {
+          delete state[n];
+        }
+        break;
+
+      case 'clear':
+        for (const ch in state) {
+          delete state[ch];
+        }
+    }
```

Comment:
It's a good practice to add default behavior. If you do not have such one - just throw an error with a text "invalid type of action"